### ISSUE TITLE
nvcc : fatal error : Cannot find compiler 'cl.exe' in PATH

### ISSUE THREAD

**Issue by Singularity00** [Theano/Theano#3815]
Can some one please help me out with these?? I followed everything from http://deeplearning.net/software/theano/install_windows.html#install-windows. But when I try to run it in GPU this error occurs.

nvcc : fatal error : Cannot find compiler 'cl.exe' in PATH
['nvcc', '-shared', '-O3', '--use-local-env', '--cl-version=2008', '-Xlinker', '/DEBUG', '-D HAVE_ROUND', '-m64', '-Xcompiler', '-DCUDA_NDARRAY_CUH=bf8d22b192850a85a00263cce1175696,-D NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION,/Zi,/MD', '-Ic:\scisoft\theano\theano\sandbox\cuda', '-IC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\numpy\core\include', '-IC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\include', '-o', 'C:\Users\USER\AppData\Local\Theano\compiledir_Windows-8-6.2.9200-AMD64_Family_21_Model_48_Stepping_1_AuthenticAMD-2.7.10-64\cuda_ndarray\cuda_ndarray.pyd', 'mod.cu', '-LC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\libs', '-LC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64', '-lpython27', '-lcublas', '-lcudart']
ERROR (theano.sandbox.cuda): Failed to compile cuda_ndarray.cu: ('nvcc return status', -1, 'for cmd', 'nvcc -shared -O3 --use-local-env --cl-version=2008 -Xlinker /DEBUG -D HAVE_ROUND -m64 -Xcompiler -DCUDA_NDARRAY_CUH=bf8d22b192850a85a00263cce1175696,-D NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION,/Zi,/MD -Ic:\scisoft\theano\theano\sandbox\cuda -IC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\lib\site-packages\numpy\core\include -IC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\include -o C:\Users\USER\AppData\Local\Theano\compiledir_Windows-8-6.2.9200-AMD64_Family_21_Model_48_Stepping_1_AuthenticAMD-2.7.10-64\cuda_ndarray\cuda_ndarray.pyd mod.cu -LC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64\libs -LC:\SciSoft\WinPython-64bit-2.7.10.3\python-2.7.10.amd64 -lpython27 -lcublas -lcudart')
WARNING (theano.sandbox.cuda): CUDA is installed, but device gpu is not available  (error: cuda unavilable)
NP time: 0.467000[s], theano time: 0.431000[s](times should be close when run on CPU!)
Result difference: 0.000000

**Comment by AlexNguyen124** [comment]
I do not specifically recall how or if I solved this for windows I find that theano works a lot better with linux and that is what i am using now to run theano.

**Comment by Singularity00** [comment]
thanks a lot. I am now going to migrate to linux. Its been days and Im still stuck with running Cuda. Regards.  

    On Tuesday, December 29, 2015 9:10 AM, AlexNguyen124 <notifications@github.com> wrote:
 

 I do not specifically recall how or if I solved this for windows I find that theano works a lot better with linux and that is what i am using now to run theano. —

**Comment by janthonywilson** [comment]
I am also working on getting theano to run on windows with GPU.  The problem is with the command line generation.  In nvcc_compiler.py around line 382 there is a "try" with a subprocess call.  Before that subprocess print "location" and ' '.join(cmd).  Then run again, and in a terminal window you can issue these commands directly (cd to "location", then copy/pasta run the nvcc command).  You will see that there is an error with a ",-D" with your "-D NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION,/Zi,/MD"  Get rid of the comma and run, it should compile.  Looking for the source of the problem now...

**Comment by bozothegrey** [comment]
Hello, are you sure is not just an issue of adding something like C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin to your PATH ?

**Comment by asap32** [comment]
Same issue complete healed on Windows 8.1 with added PATH as suggested by  bozothegrey.

**Comment by Superraptor** [comment]
I'm not positive if the same exact issue, but simply adding the above PATH on a Windows 10 machine did not work (at least in my specific situation).

**Comment by hunter-ccri** [comment]
I followed instructions I found elsewhere online to download and mount MSVS 15 to enable installation of CUDA 8 (which is now supported by MSVS 15). Mounting MSVS 15 doesn't create a /VS/bin/cl.exe anywhere I can find, so I specified --cl-version=2014 and MSVS 2014 /VS/bin as nvcc.compile_dir.  It doesn't recognize the key for 2014. Specifying 2015 it does not throw this error but cannot find compiler cl.exe - so maybe I need to download and install MSVS 2015?
